virtinst/domain/os.py

```diff
@@ -79,12 +79,26 @@ def is_riscv_virt(self):
         return self.is_riscv() and str(self.machine).startswith("virt")
 
     XML_NAME = "os"
-    _XML_PROP_ORDER = ["arch", "os_type", "loader", "loader_ro", "loader_type",
-                       "nvram", "nvram_template", "kernel", "initrd",
-                       "initdir", "inituser", "initgroup",
-                       "kernel_args", "dtb", "bootdevs", "smbios_mode"]
+    _XML_PROP_ORDER = [
+            "os_type", "arch", "machine",
+            "loader", "loader_ro", "loader_type", "loader_secure",
+            "firmware", "firmware_features", "nvram", "nvram_template",
+            "bootdevs", "bootmenu_enable", "bootmenu_timeout",
+            "bios_useserial", "bios_rebootTimeout", "smbios_mode",
+            "kernel", "initrd", "kernel_args", "dtb",
+            "acpi_tb", "acpi_tb_type",
+            "init", "initargs", "initenvs", "initdir", "inituser", "initgroup"]
```

Comment:
I'm okay with reordering things, and comments etc, but if you are going to change options here, it should be to match them with libvirt ordering. Check libvirt [src/conf/domain_conf.c](https://gitlab.com/libvirt/libvirt/-/raw/master/src/conf/domain_conf.c) and search for `virBufferAddLit(buf, "<os");` which is the start of where the `<os>` block is written, follow the order that values are written, which is the order they will show up in the XML, and have our output match that. It's not required for any reason but it makes it easier to diff virt-install XML against libvirt formatted XML which helps in some debugging scenarios

Reply:
Thanks for the tip! Bringing it in line with the XML order was part of the goal, except I didn't know about `domain_conf.c` and ordered mostly as they are presented in the [Domain XML docs](https://libvirt.org/formatdomain.html#operating-system-booting), with a few exceptions.

I've brought it in line with the XML ordering now and squashed those changes into the cleanup/reordering commit.